src/components/CreateDeliveryModal.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { addDelivery } from "../fake-db";
+
+const CreateDeliveryModal = (props: {
+  x: boolean;
+  setX: (x: boolean) => void;
+}) => {
+  const { x, setX } = props;
+  const queryClient = useQueryClient();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { mutate } = useMutation({
+    mutationFn: (opts: {
+      name: string;
+      status: string;
+      deliveryDate: Date;
+    }) => addDelivery(opts.name, opts.status, opts.deliveryDate as any as string),
+    onMutate: () => {
+      setIsLoading(true);
+    },
+    onSuccess: async () => {
+      console.log("Invalidating query");
+      queryClient.invalidateQueries({ queryKey: ["deliveries"] });
+      setIsLoading(false);
+    },
+  });
+
+  const [name, setName] = React.useState("");
+  const [status, setStatus] = useState("");
+  const [deliveryDate, setDeliveryDate] = React.useState("");
+
+  return (
+    <div className="absolute bg-white top-10 left-10 bottom-10 right-10 shadow-xl border-2 border-black">
+      <h1 className="text-2xl my-2 flex flex-row">
+        <span className="flex-grow">Create delivery</span>
+        <button className="btn" onClick={() => setX(false)}>
+          X
+        </button>
+      </h1>
```

Comment:
this should not be an `h1` since there already is one on this page. it should also not wrap the button since that's not semantically correct
```suggestion
      <div className="flex flex-row justify-between my-2">
        <h2 className="text-2xl">Create delivery</h2>
        <button className="btn" onClick={() => setX(false)}>
          X
        </button>
      </div>
```